src/app/script/canvas_widget.cpp

```diff
@@ -69,6 +69,26 @@ void Canvas::onInitTheme(ui::InitThemeEvent& ev)
   setBgColor(bg);
 }
 
+static std::unique_ptr<ui::MouseMessage> makeScaled(ui::MouseMessage* mouseMsg, const gfx::Point& offset)
+{
+  std::unique_ptr<ui::MouseMessage> scaledMouseMsg = std::make_unique<ui::MouseMessage>(
+                                     mouseMsg->type(),
+                                     mouseMsg->pointerType(),
+                                     mouseMsg->button(),
+                                     mouseMsg->modifiers(),
+                                     ((mouseMsg->position()-offset)/ui::guiscale())+offset,
+                                     mouseMsg->wheelDelta(),
+                                     mouseMsg->preciseWheel(),
+                                     mouseMsg->pressure());
+  scaledMouseMsg->setRecipient(mouseMsg->recipient());
+  scaledMouseMsg->setCommonAncestor(mouseMsg->commonAncestor());
+  scaledMouseMsg->setDisplay(mouseMsg->display());
+  scaledMouseMsg->setPropagateToChildren(mouseMsg->propagateToChildren());
+  scaledMouseMsg->setPropagateToParent(mouseMsg->propagateToParent());
+  scaledMouseMsg->setFromFilter(mouseMsg->fromFilter());
+  return scaledMouseMsg;
+}
+
```

Comment:
There is a ctor to copy a MouseMessage with a different position (also I've changed the return type to a value instead of a pointer):
```suggestion
static ui::MouseMessage makeScaled(ui::MouseMessage* mouseMsg, const gfx::Point& offset)
{
  return ui::MouseMessage(
    mouseMsg->type(), *mouseMsg,
    ((mouseMsg->position() - offset) / ui::guiscale()) + offset);
}
```

Reply:
@dacap Okay I have just tested this and the issue here is that the recipient is not copied (this crashes the program when the mouse message is received). Should I add it to the constructor code or should I assign the recipient after creating the copy (in makeScaled function)?

Reply:
Copy the receipt and display, don't add them in the ctor. Search for "mouseMsg2" in the source code, in other occasions this is done (and sometimes the receipt or the display changes).